src/components/Auth/UserFormDialog/UserForm.test.tsx

```diff
@@ -30,93 +37,117 @@ describe('UserForm', () => {
-    expect(onSave).toHaveBeenCalledWith(
-      jasmine.objectContaining({
+    expect(updateUser).toHaveBeenCalledWith({
+      localId: user.localId,
+      user: jasmine.objectContaining({
         displayName,
         phoneNumber,
-      })
-    );
+      }),
+    });
+    expect(clearAuthUserDialogData).toHaveBeenCalled();
   });
 
-  it('calls onSave, but does not close the form when "create and new" clicked.', async () => {
-    const phoneNumber = '+1 555-555-0100';
+  it('calls onCreate on form submit if user is not provided', async () => {
     const {
+      triggerValidation,
       getByText,
       getByLabelText,
-      triggerValidation,
-      onSave,
-      onClose,
-    } = setup({
-      displayName: '',
-    });
+      createUser,
+      clearAuthUserDialogData,
+    } = setup({});
+
+    expect(createUser).not.toHaveBeenCalled();
+
+    getByText('Add a user');
 
     const input = getByLabelText(/Phone/) as HTMLInputElement;
 
-    fireEvent.change(input, {
-      target: { value: phoneNumber },
+    await act(async () => {
+      await fireEvent.change(input, {
+        target: { value: phoneNumber },
+      });
+      await fireEvent.blur(input);
     });
-    fireEvent.blur(input);
 
     await triggerValidation();
+
+    expect(createUser).toHaveBeenCalled();
+    expect(clearAuthUserDialogData).toHaveBeenCalled();
+  });
+
+  it('calls onCreate, but does not close the form when "create and new" clicked', async () => {
+    const {
+      triggerValidation,
+      getByLabelText,
+      getByText,
+      createUser,
+      clearAuthUserDialogData,
+    } = setup({});
+
+    expect(createUser).not.toHaveBeenCalled();
+
+    const input = getByLabelText(/Phone/) as HTMLInputElement;
+
     await act(async () => {
-      await fireEvent.click(getByText('Save and create another'));
+      await fireEvent.change(input, {
+        target: { value: phoneNumber },
+      });
+      await fireEvent.blur(input);
     });
-    await triggerValidation();
 
-    // Resets the form
-    expect(input.value).toBe('');
-    // Creates user
-    expect(onSave).toHaveBeenCalledWith(
-      jasmine.objectContaining({
-        phoneNumber: phoneNumber,
-      })
-    );
-  });
+    await triggerValidation();
 
-  it('sets input values based on passed user value', async () => {
-    const email = 'lol@lol.lol';
-    const password = 'qwerty';
-    const { getByLabelText, queryByLabelText } = setup({
-      displayName,
-      email,
-      password,
+    await act(async () => {
+      await fireEvent.click(getByText('Save and create another'));
     });
-    expect((getByLabelText(/Display name/) as HTMLInputElement).value).toBe(
-      displayName
-    );
-    expect((queryByLabelText('Email') as HTMLInputElement).value).toBe(email);
-    expect((queryByLabelText('Password') as HTMLInputElement).value).toBe(
-      password
-    );
+
+    expect(createUser).toHaveBeenCalled();
+    expect(clearAuthUserDialogData).toHaveBeenCalled();
```

Comment:
Please add an assertion on the dialog still open

Reply:
Also: there are reports of the dialog got closed in the bugbash. Please investigate

Reply:
I moved the dialog moving logic out of the form and fixed the bug where the dialog closed.